salt/fileclient.py

```diff
@@ -461,16 +462,23 @@ def get_url(self, url, dest, makedirs=False, saltenv='base',
         Get a single file from a URL.
         '''
         url_data = urlparse(url)
+        url_scheme = url_data.scheme
+        url_path = os.path.join(
+                url_data.netloc, url_data.path).rstrip(os.sep)
 
-        if url_data.scheme in ('file', ''):
+        if url_scheme is not '' and url_scheme in string.letters:
```

Comment:
The first half of conditional is not needed here and should be removed because it is implied by the second. Also, `letters` is not an attribute of the `string` module in Python 3 so this will completely break Salt's Py3 support.